default.nix

```diff
@@ -630,12 +630,27 @@ in let this = rec {
     buildDepends = (drv.buildDepends or []) ++ generalDevTools (nativeHaskellPackages haskellPackages);
   })).env;
 
-  workOnMulti' = { env, packageNames, tools ? _: [] }:
+  workOnMulti' = { env, packageNames, tools ? generalDevTools }:
     let ghcEnv = env.ghc.withPackages (packageEnv: builtins.concatLists (map (n: (packageEnv.${n}.override { mkDerivation = x: { out = builtins.filter (p: builtins.all (nameToAvoid: (p.pname or "") != nameToAvoid) packageNames) ((x.buildDepends or []) ++ (x.libraryHaskellDepends or []) ++ (x.executableHaskellDepends or []) ++ (x.testHaskellDepends or [])); }; }).out) packageNames));
+
+        # mergeOnDrvName combines two funcs that generate lists of derivations
+        # dropping any derivations created by the second whose name is the same
+        # as one of the derivations generated by the first
+        mergeOnDrvName = mkXs: mkYs: toolEnv:
```

Comment:
I don't think this is necessary. Having the same derivation in `buildInputs` multiple times is fine. I'd leave it out.

Reply:
Actually, I added this to address the concern you mention above, about tools being lost when the tools option is specified - I'm sorry if that's not clear.  Perhaps I should update comment to clarify this?

The function is used to merge the user-provided tools with generalDevTools, preferring derivations from the former if there is a conflict.

There should be no case when tools are lost; e.g. if you specify `{...; tools _:  []; ..}`, you'll get all the tools specified by generalDevTools.

Reply:
Oh! I missed that completely, sorry.

In that case, it might make more sense to switch from a list to an attr set? Keys are the names that we want to be unique and values are the derivations. That way this just becomse `//`

Reply:
I'll give that a try and update the PR.

Reply:
Done - the function mergeOnDrvName is still present, but is simpler as it is implemented using attrSets and `//`.

Reply:
Oh I was actually thinking about having `generalDevTools` and the `tools` option use attr sets instead of lists. Would avoid the whole problem. I guess it has the problem of not being backwards-compatible... But at least it would be an obvious type error? @3noch Curious what you think.

Reply:
Keeping backwards compatibility seems ideal in this case especially since having sets with null keys is kinda ugly. Can't we just concat `generalDeTools` in all cases and make default tools `[]`?

Reply:
@3noch That's basically what we do now. The problem is when a user wants to change e.g. what cabal-install version is provided. Not sure what you mean by null keys.

Reply:
So - may I leave this PR as-is?  At the moment, the change 
- ensures that user-selected versions of tools are preferred to the versions of tools in generalDevTools like cabal-install
- is backwards-compatible; it should not affect any existing users; they always get generalDevTools